File: LeetCode/169/169.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    Solution(){}
    ~Solution(){}
    int majorityElement(vector<int>& nums) {
        int coor = -1;
        int count = 0;
        for(int num: nums) {
            if (count == 0)     coor = num;
            count += (coor == num ? 1 : -1);
        }
        cout << coor;
        return coor;
    }
};
int main(int argc, char const *argv[])
{
    vector<int> nums;
    nums.push_back(1);
    nums.push_back(3);
    nums.push_back(4);
    nums.push_back(5);
    nums.push_back(3);
    nums.push_back(3);
    nums.push_back(3);    
    Solution S;
    S.majorityElement(nums);
    return 0;
}

File: 排序/例题/LC169-多数元素/169.py
# from collections import Counter
# class Solution:
#     def majorityElement(self, nums: List[int]) -> int:
#         return Counter(nums).most_common(1)[0][0]

class Solution(object):
	def majorityElement(self, nums):
		"""
		:type nums: List[int]
		:rtype: int
		"""
		corr = None
		count = 0
		for num in nums:
			if count == 0:	corr = num
			count += (1 if num == corr else -1)
		# print(corr)
		return corr
if __name__ == "__main__":
	s = Solution()
	s.majorityElement([1, 2, 3, 3, 4, 3, 3])